search/double_linear_search.cpp

```diff
@@ -0,0 +1,58 @@
+#include <iostream>
+#include <vector>
+#include <numeric>
+#include <cassert>
+
+/*--------------------------------------------------------------------------
+/Iterate through the array from both sides to find the index of search_item.
+/
+/:param array: the array to be searched
+/:param search_item: the item to be searched
+/:return the index of search_item, if search_item is in array, else -1
+/
+/ Examples:
+/ >>> double_linear_search([1, 5, 5, 10], 1)
+/ 0
+/ >>> double_linear_search([1, 5, 5, 10], 5)
+/ 1
+/ >>> double_linear_search([1, 5, 5, 10], 100)
+/ -1
+/ >>> double_linear_search([1, 5, 5, 10], 10)
+/ 3
+/--------------------------------------------------------------------------*/
+int double_linear_search(std::vector<int> list, int search_item) {
```

Comment:
Functions should be documented explaining the use of its parameters, a brief description, and what it returns.